app/models/procedure.rb

```diff
@@ -49,6 +49,9 @@ class Procedure < ApplicationRecord
   has_one :protocols_participant, through: :appointment
   has_one :visit_group, through: :appointment
 
+  delegate :set_deletable_flag, to: :protocols_participant
+
+  after_initialize :set_deletable_flag
```

Comment:
`after_initialize` isn't very helpful here.  That method will trigger when a new object is instantiated like in `Procedure.new` or when an object is loaded like in `@procedure = Procedure.find(22484)`.  Neither of which would be helpful since you want the check to run after there is some change to an object.  `after_update` might would probably be a better method to use here.